nvme-print: print always celsius temperature value

ikegami-t opened the pull request:
The fahrenheit value also printed if used by the locale country.

ikegami-t commented:
Fixes for the issue #2910 and #2456 as below for example.
1. locale country: US (used fahrenheit)
- stdout ouput
```sh
root@tokunori-X570-Taichi:/home/tokunori/nvme-cli# .build/nvme smart-log /dev/nvme0
...
temperature                             : 41 °C (314 K, 105 °F)
...
Temperature Sensor 1                    : 41 °C (314 K, 105 °F)
Temperature Sensor 2                    : 50 °C (323 K, 122 °F)
...
```
- json output
```sh
root@tokunori-X570-Taichi:/home/tokunori/nvme-cli# .build/nvme get-feature /dev/nvme1 -o json | grep TM
      "Temperature Threshold Hysteresis (TMPTHH)":"-273 Celsius",
      "TMPTHH kelvin":"0 K",
      "TMPTHH fahrenheit":"-459 Fahrenheit",
      "Threshold Temperature Select (TMPSEL)":0,
      "TMPSEL description":"Composite Temperature",
      "Temperature Threshold (TMPTH)":"77 Celsius",
      "TMPTH kelvin":"350 K",
      "TMPTH fahrenheit":"170 Fahrenheit"
      "Thermal Management Temperature 1 (TMT1)":"344 K",
      "TMT1 celsius":"71 Celsius",
      "TMT1 fahrenheit":"159 Fahrenheit",
      "Thermal Management Temperature 2 (TMT2)":"348 K",
      "TMT2 celsius":"75 Celsius",
      "TMT2 fahrenheit":"167 Fahrenheit"
```
2. locale country: JP (not used fahrenheit)
- stdout ouput
```sh
tokunori@tokunori-X570-Taichi:~/nvme-cli$ nvme-build smart-log /dev/nvme0
...
temperature                             : 41 °C (314 K)
...
Temperature Sensor 1                    : 41 °C (314 K)
Temperature Sensor 2                    : 50 °C (323 K)
...
```
- json output
```sh
tokunori@tokunori-X570-Taichi:~/nvme-cli$ nvme-build get-feature /dev/nvme1n1 -o json | grep TM
      "Temperature Threshold Hysteresis (TMPTHH)":"-273 Celsius",
      "TMPTHH kelvin":"0 K",
      "Threshold Temperature Select (TMPSEL)":0,
      "TMPSEL description":"Composite Temperature",
      "Temperature Threshold (TMPTH)":"77 Celsius",
      "TMPTH kelvin":"350 K"
      "Thermal Management Temperature 1 (TMT1)":"344 K",
      "TMT1 celsius":"71 Celsius",
      "Thermal Management Temperature 2 (TMT2)":"348 K",
      "TMT2 celsius":"75 Celsius"
```

ikegami-t commented on nvme-print-json.c:
```diff
@@ -3504,6 +3504,17 @@ static void json_feature_show_fields_lba_range(struct json_object *r, __u8 field
+	char json_str[STR_LEN];
+
+	if (fahrenheit) {
+		sprintf(json_str, "%s", fahrenheit);
```
Just fixed as mentioned. Thank you!

wdoekes commented:
Thanks for picking this up :+1:  Note that you're not yet fixing the setlocale (ab)use I mention in #2910

wdoekes commented:
> About the build error fixed by the commit: https://github.com/linux-nvme/nvme-cli/commit/9444ce0. (Sorry for the error.)

Oh. In that case, all good. Carry on!

----

> Anyway really required all three values to print I will fix the commit again as so.

I don't care about the conditionality now, as long as Celcius stays where it is :)

> Still looks the current nvme-cli LC_MEASUREMENT usecase not invalid for me.

It "works", but its usage is wrong. You're violating how the locale functions should work:
- by hardcoding language properties (`"US", ...`)
- instead of using the framework as it exists (`_NL_MEASUREMENT_MEASUREMENT`)

I tried to explain how it "should be properly implemented" in https://github.com/linux-nvme/nvme-cli/issues/2456#issuecomment-3307263950

On the other hand, with the 9444ce0 fix, your version is now more compatible between glibc/musl than any fix using `_NL_MEASUREMENT_MEASUREMENT` would be, so there's that.

_Like I said. Not a hill I'm willing to die on._

ikegami-t commented:
Thanks for your review comment! (Sorry for the error.) Just fixed the patch to check the LC_MEASUREMENT to build.

ikegami-t commented:
Fixed as mentioned then always printed all the celsius and kelvin, fahrenheit values also so no locale information checked. And removed the json changes also. Thank you.
